lib/client/client.go

```diff
@@ -93,17 +93,23 @@ func (proxy *ProxyClient) GetSites() ([]services.Site, error) {
 
 	stdout := &bytes.Buffer{}
 	reader, err := proxySession.StdoutPipe()
-	done := make(chan struct{})
+	done := make(chan error)
 	go func() {
-		io.Copy(stdout, reader)
+		_, err := io.Copy(stdout, reader)
+		if err != nil && err != io.EOF {
+			done <- err
+		}
 		close(done)
 	}()
 
 	if err := proxySession.RequestSubsystem("proxysites"); err != nil {
 		return nil, trace.Wrap(err)
 	}
 	select {
-	case <-done:
+	case doneErr, ok := <-done:
+		if ok && doneErr != nil {
+			return nil, trace.Errorf("read err: %s", doneErr)
```

Comment:
It infers it automatically from its call stack. Just call `trace.Wrap(err)` - we always wrap errors. And it's ok to double or triple-wrap (it's smart enough to not do that)